src/main/java/org/springframework/up/config/TemplateRepositoryProperties.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.up.config;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ * @author Thomas Risberg
+ */
+@ConfigurationProperties("up.git")
+public class TemplateRepositoryProperties {
+
+	/*
+	 * Map containing the access tokens for Git repositories. The key should be the Git
+	 * repository host like 'github.com' and the value should be the corresponding access
+	 * token.
+	 */
+	private Map<String, String> tokens = new HashMap<>();
```

Comment:
We don't want to keep tokens in a spring config(as those would get baked in to an app). I did some work to have cli related config files in users home which login/logout commands would update. That's how github cli stores users tokens.